Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ClientGameNetPortal.cs

```diff
@@ -234,6 +235,7 @@ void ConnectClient()
             //  If the socket connection fails, we'll hear back by getting an ReceiveServerToClientSetDisconnectReason_CustomMessage callback for ourselves and get a message telling us the reason
             //  If the socket connection succeeds, we'll get our ReceiveServerToClientConnectResult_CustomMessage invoked. This is where game-layer failures will be reported.
             m_Portal.NetManager.StartClient();
+            SceneLoaderWrapper.Instance.AddOnSceneEventCallback();
```

Comment:
1. ClientGameNetPortal shouldn't have to know about this, it's not its responsibility and just adds coupling.
2. method name is a bit weird, add...something usually means you're adding something and would expect a parameter. "Init" something might be better.

Reply:
The issue with this is that to trigger the loading screen at the beginning of client syncrhonization, we need to register those callbacks before they happen. However, the NetworkSceneManager is created when StartClient / StartHost / StartServer is called, so we have to register them after this. OnClientConnected and OnNetworkSpawn happen after the synchronization has finished so it must be done before those callbacks. We currently don't have access to callbacks when starting clients, which is why I had to do it this way.

Reply:
I'd create our own callback for this then, instead of having our connection management know about scene management initialization?

Reply:
If we do that though, that means that we need to add coupling in the other direction, which means we will have to move the loading screen stuff out of the Utilities package. My initial goal was to create something generic enough to be able to be reused in other projects, but if it makes more sense that way, I can move it.

Reply:
@pdeschain you have the arch cleanup initiative. What do you think of moving connection management to Utilities as well? To me it's the kind of stuff we'd want to reuse in future samples anyway

Reply:
Should that be part of a future ticket instead? Or should we move all of that to the Shared assembly for now and evaluate later what should be moved to Utilities?

Reply:
future ticket, just make sure it's tracked :)

Reply:
here it is: https://jira.unity3d.com/browse/MTT-3157